### DISCUSSION TITLE
headerAction to use current filtered query?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#14448] [Help] [answered]
### Package

Table builder

### Package Version

v3.2.113

### How can we help you?

I have a resource table, with headerActions. 

When I pass on the `$table` and use `$table->getQuery()` I get the default query.

```
Tables\Actions\Action::make('export')
  ->action(function(Table $table) {
    $query = $table->getQuery(); // this is the default query, without the filters

    dd($query->count()); // all records
  }),
```

How do I get the current query **including** the active filters?

**Comment by dissto** [comment]
Maybe [getFilteredTableQuery](https://github.com/filamentphp/filament/blob/3.x/packages%2Ftables%2Fsrc%2FConcerns%2FHasRecords.php#L24) ? 🤔

**Reply by wivaku** [reply]
`$query =  $table->getFilteredTableQuery()` results in:
```
Method Filament\Tables\Table::getFilteredTableQuery does not exist.
```

**Reply by dissto** [reply, marked as answer]
The above function is probably on `$livewire`

**Reply by wivaku** [reply]
You're right, thanks!

**Comment by wivaku** [comment]
Full answer based on @dissto's suggestion:

```
Tables\Actions\Action::make('export')
  ->action(function($livewire) {
    $query = $livewire-> getFilteredTableQuery();

    dd($query->count()); // filtered records
  }),
```
